unyt/exceptions.py

```diff
@@ -47,6 +47,29 @@ def __str__(self):
         return err
 
 
+class UnitOperationWarning(UserWarning):
+    """A warning that is raised when unit operations are not allowed but
+    when running with a UnitRegistry with strict=False. In this case,
+    the operation is allowed to continue but with unit information stripped.
+    """
+
+    def __init__(self, operation, unit1, unit2=None):
+        self.operation = operation
+        self.unit1 = unit1
+        self.unit2 = unit2
+        UserWarning.__init__(self)
+
+    def __str__(self):
+        err = (
+            'The %s operator for unyt_arrays with units "%s" '
+            '(dimensions "%s") ' % (self.operation, self.unit1, self.unit1.dimensions)
```

Comment:
in reprs and error messages, `'` is the preferred delimiter for strings, over `"`. `!r` ensures that we use the raw string repr (`'string'`), and we can use fstrings too now !
```suggestion
            f"The {self.operation.__name__} operator for unyt_arrays"
            f" with units {self.unit1!r} (dimensions {self.unit1.dimensions!r})"
```